base/utferror.jl

```diff
@@ -0,0 +1,30 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+##\brief      Error messages for Unicode / UTF support
```

Comment:
There was this comment (from Stefan):
>Seems like once the doc string format settles down we could teach doxygen to parse it and then use doxygen for everything.

This:
>+1. Having better-documented internals will have countless benefits. In particular, I expect that documenting the ownership of various pointers/handles will improve (or at least help us better assess) memory leaks going forward.

> I do think Julia's a large enough project now with enough people regularly trying to look at and work with the internals that it's past time for that to be a good enough reason to resist documenting things.

There are a number of others that are positive, even about having the doxygen format:
>I've personally rarely found the actual output of Doxygen to be very useful, but I have found that having Doxygen-styled comments in a code base often leads to an improvement in the overall documentation level of the code.

It seems to me that the objections boil down to:
1) This seems fine, but, it's not our standard yet.
2) I haven't seen output from doxygen that I found useful or that I liked.
A third argument, that I heard in a previous life, that thankfully I haven't seen here is
3) I just can't be bothered to add documentation

To address 1, I'd say that you have to start somewhere, and having this documentation in the code that I am adding would at least give other Julia contributors time to evaluate the format, good or bad.  All that really remains here is 1-2 extra characters per line, i.e. `##\` instead of just `#`, or `##<` instead of just `#`.  Stripping them out at a later date if it is decided people don't like them would be a trivial regex search-replace task, but adding them back in if they are stripped out now would involve time and effort.
About the second argument, well, a number of people have already commented that they found *some* use out of the doxygen output, if only just as a cross-referencing tool (I use it effectively for that, and to find documentation of *internal* functions and structures, and structure members).
Just because you haven't found value out of it, doesn't mean that it should be denied to other people who do find value in it.
And the third argument (which might be implicit in the lack of documentation I've found in Julia sources), well, nobody is forcing you to write documentation... don't prevent other people who *are* willing to spend time documentating the code from using what they feel to be an effective tool (although not perfect)

If you look, pretty much *all* the documentation generators use pretty much the same syntax, derived from javadoc.  The reasons I felt doxygen in particular would be a good fit for Julia
(besides my personal familiarity with it) are:
1) It supports many languages, C/C++, C#, Fortran, Java, JavaScript, Lua, Python etc.
    (I think it may be the only documentation system that supports all the languages I have seen used to build Julia)
2) It supports using Markdown in the documentation, so it fits well with the Docile
    `@doc` format.
3) Its support of Python docstrings and `#` style macros mean that not much would need to be done to make it support Julia as well.